### PR TITLE
Add tests to ci

### PR DESCRIPTION
**Pull request by peterfication**
Run the tests in CI.

The way to run the tests is taken from [folke/noice.nvim](https://github.com/folke/noice.nvim/blob/main/tests/init.lua).

### CODE REVIEW

**Comment by johmsalas** [comment]
Tests are running :partying_face: 

Screenshot_20231110_194900

**Review by johmsalas — APPROVED** [review]
Thanks! :medal_sports:

**Comment by peterfication** [comment]
A rebase was needed because it was conflicting with the StyLua PR of course.

I can just merge as you already approved it even though a new commit was added. I would suggest to adjust the branch protection rules:
- Require a pull request before merging 
  - Dismiss stale pull request approvals when new commits are pushed 

What do you think @johmsalas?

After this PR is merged it probably makes sense to 
- Require status checks to pass before merging (StyLua and Tests (after this PR is merged))

**Pull request merged**
